### PR TITLE
[no-release-notes] simplified warn on error behavior

### PR DESCRIPTION
**Pull request by zachmu**
Previous behavior had many places where it checked the warn on error setting, but only one is actually necessary (StartTransaction). We should probably get rid of that setting altogether soon.

Also added a test that tag deletions are properly replicated.

### CODE REVIEW

File: integration-tests/bats/replication.bats [hunk 1]
```diff
@@ -108,6 +108,35 @@ teardown() {
     [[ ! "$output" =~ "feature" ]] || false
 }
 
+@test "replication: push tag delete, pull delete on read" {
```
Comment (max-hoffman) [inline integration-tests/bats/replication.bats:111]: just double checking that you meant to include this

**Review by max-hoffman — APPROVED** [review]
lgtm assuming we have some duplicate mechanism catching the warnings, idk what relation the bats test has to these changes also

**Comment by coffeegoddd** [comment]
@zachmu __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.997743 to 99.997743</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>f5df985</TD>
<TD>not ok</TD>
<TD>134</TD>
</TR>
<TR><TD>f5df985</TD>
<TD>ok</TD>
<TD>5937323</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>f5df985</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.997743</TD>
</TR>
</table>

File: integration-tests/bats/replication.bats (hunk 1 above)
Comment (zachmu) [reply integration-tests/bats/replication.bats:111]: Yeah, it was missing from my first PR

Comment (zachmu) [reply integration-tests/bats/replication.bats:111]: Yeah it was missing from my first PR

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.997743 to 99.997743</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>da1bb8d</TD>
<TD>not ok</TD>
<TD>134</TD>
</TR>
<TR><TD>da1bb8d</TD>
<TD>ok</TD>
<TD>5937323</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>da1bb8d</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.997743</TD>
</TR>
</table>

**Pull request merged**